adapters/adgeneration/adgeneration.go

```diff
@@ -112,12 +116,32 @@ func (adg *AdgenerationAdapter) getRawQuery(id string, request *openrtb2.BidRequ
 	if adSize != "" {
 		v.Set("sizes", adSize)
 	}
+	if request.Device != nil && request.Device.OS == "android" {
+		v.Set("sdktype", "1")
+	} else if request.Device != nil && request.Device.OS == "ios" {
+		v.Set("sdktype", "2")
+	} else {
+		v.Set("sdktype", "0")
+	}
 	if request.Site != nil && request.Site.Page != "" {
 		v.Set("tp", request.Site.Page)
 	}
 	if request.Source != nil && request.Source.TID != "" {
 		v.Set("transactionid", request.Source.TID)
```

Comment:
This line isn't new code, but I checked the test coverage and it isn't covering this case, so we'll need a json test for this too!

Reply:
@AlexBVolcy 
Thank you. I fixed it.